components/server/src/orgs/organization-service.ts

```diff
@@ -38,45 +41,159 @@ export class OrganizationService {
         return result;
     }
 
-    /**
-     * getGenericInvite returns the generic invite for the given organization or creates one if no invite exists.
-     *
-     * @param userId
-     * @param orgID
-     * @returns
-     */
-    public async getOrCreateInvite(userId: string, orgID: string): Promise<TeamMembershipInvite> {
-        await this.checkPermissionAndThrow(userId, "invite_members", orgID);
-        const invite = await this.teamDB.findGenericInviteByTeamId(orgID);
+    public async deleteOrganization(userId: string, orgId: string): Promise<void> {
+        await this.checkPermissionAndThrow(userId, "delete", orgId);
+        const projects = await this.projectsService.getProjects(userId, orgId);
+
+        const members = await this.teamDB.findMembersByTeam(orgId);
+        try {
+            await this.teamDB.transaction(async (db, ctx) => {
+                for (let project of projects) {
+                    await this.projectsService.deleteProject(userId, project.id, ctx);
+                }
+                for (let member of members) {
+                    await db.removeMemberFromTeam(member.userId, orgId);
+                }
```

Comment:
As a follow-up, would it be possible to remove all members at once instead of the iteration?